test/manual/hint_test.cc

```diff
@@ -44,6 +44,7 @@
 #include "db/commitlog/rp_set.hh"
 #include "log.hh"
 #include "schema.hh"
+#include "db/hints/sync_point.hh"
```

Comment:
Why test/manual? Can it not run as part of the automated suite?

Reply:
You are right, it should be automatic. I put the new test in this file because it was the only file with hinted handoff unit tests. I'm not sure why the existing test is not run automatically, I'll do some git-archaeology and try to find out why it is the case.

Reply:
@vladzcloudius it looks like the `hint_test.cc` was never run automatically - although it was added to `configure.py` so that it is compiled (see [here](https://github.com/scylladb/scylla/commit/b20dbe16d84a3c1d1c77de90ee3cff6bbf2dc044)), it wasn't added to the list of test that should be run automatically in `test.py` (see [here](https://github.com/scylladb/scylla/blob/b20dbe16d84a3c1d1c77de90ee3cff6bbf2dc044/test.py#L30-L101)). Do you remember if it was intentional? It doesn't seem to be, so I'll preemptively move it to automatic tests in this PR - but I can move it back if you object.

Reply:
You can add a new test file in the automatic section, and later move the manual test. This way we don't add a new blocker.

Reply:
Done, added another `hint_test.cc` file in `test/boost/` and placed the new test there.